SimplifiedTests/NYPLAgeCheckTests.swift

```diff
@@ -0,0 +1,127 @@
+    ageCheck.verifyCurrentAccountAgeRequirement(userAccountProvider: userAccountProviderMock,
+                                                currentLibraryAccountProvider: simplyeLibraryAccountProviderMock) { [weak self] (aboveAgeLimit) in
+      XCTAssertFalse(aboveAgeLimit)
+      XCTAssertTrue(self?.ageCheckChoiceStorageMock.userPresentedAgeCheck ?? false)
+      self?.expectation.fulfill()
+    }
+    
+    let birthYear = Calendar.current.component(.year, from: Date()) - 12
+    ageCheck.didCompleteAgeCheck(birthYear)
+    waitForExpectations(timeout: 1, handler: nil)
+  }
+
+  func testAge13() throws {
+    ageCheck.verifyCurrentAccountAgeRequirement(userAccountProvider: userAccountProviderMock,
+                                                currentLibraryAccountProvider: simplyeLibraryAccountProviderMock) { [weak self] (aboveAgeLimit) in
+      XCTAssertFalse(aboveAgeLimit)
+      XCTAssertTrue(self?.ageCheckChoiceStorageMock.userPresentedAgeCheck ?? false)
+      self?.expectation.fulfill()
+    }
+    
+    let birthYear = Calendar.current.component(.year, from: Date()) - 13
+    ageCheck.didCompleteAgeCheck(birthYear)
+    waitForExpectations(timeout: 1, handler: nil)
+  }
+  
+  func testAge14() throws {
+    ageCheck.verifyCurrentAccountAgeRequirement(userAccountProvider: userAccountProviderMock,
+                                                currentLibraryAccountProvider: simplyeLibraryAccountProviderMock) { [weak self] (aboveAgeLimit) in
+      XCTAssertTrue(aboveAgeLimit)
+      XCTAssertTrue(self?.ageCheckChoiceStorageMock.userPresentedAgeCheck ?? false)
+      self?.expectation.fulfill()
+    }
+    
+    let birthYear = Calendar.current.component(.year, from: Date()) - 20
+    ageCheck.didCompleteAgeCheck(birthYear)
+    waitForExpectations(timeout: 1, handler: nil)
+  }
+  
+  func testAge100() throws {
+    ageCheck.verifyCurrentAccountAgeRequirement(userAccountProvider: userAccountProviderMock,
+                                                currentLibraryAccountProvider: simplyeLibraryAccountProviderMock) { [weak self] (aboveAgeLimit) in
+      XCTAssertTrue(aboveAgeLimit)
+      XCTAssertTrue(self?.ageCheckChoiceStorageMock.userPresentedAgeCheck ?? false)
+      self?.expectation.fulfill()
+    }
+    
+    let birthYear = Calendar.current.component(.year, from: Date()) - 100
+    ageCheck.didCompleteAgeCheck(birthYear)
+    waitForExpectations(timeout: 1, handler: nil)
+  }
+  
+  func testAgeCheckFailed() throws {
+    // Use an inverted expectation to make sure the completion closure is not executed
+    self.expectation.isInverted = true
+    ageCheck.verifyCurrentAccountAgeRequirement(userAccountProvider: userAccountProviderMock,
+                                                currentLibraryAccountProvider: simplyeLibraryAccountProviderMock) { [weak self] (aboveAgeLimit) in
+      self?.expectation.fulfill()
+    }
+    
+    ageCheck.didFailAgeCheck()
+    waitForExpectations(timeout: 3, handler: nil)
```

Comment:
nice use of inverted expectations! question, does this timeout then mean that this test will always take 3 seconds to run if it succeeds? If so just wondering if we can lower it a bit.

Reply:
It will succeed if the expectation is not fulfilled by the end of the timeout. Sure I can lower it, nothing here really takes time to process so it should be fine.